safe/definitionsv4/minimum_needs.py

```diff
@@ -0,0 +1,140 @@
+# coding=utf-8
+
+"""Definitions relating to minimum-needs."""
+import re
+
+import sys
+from PyQt4.QtCore import QVariant
+
+from safe.common.parameters.resource_parameter import ResourceParameter
+from safe.definitionsv4.fields import default_field_length
+from safe.gui.tools.minimum_needs.needs_profile import NeedsProfile
+
+
+__copyright__ = "Copyright 2016, The InaSAFE Project"
+__license__ = "GPL version 3"
+__email__ = "info@inasafe.org"
+__revision__ = '$Format:%H$'
+
+
+"""
+Minimum needs definition are taken from user.
+Because of this, minimum needs field definition needs to be generated
+dynamically to avoid namespace clash. It is also need to be able to link to
+original-user-defined definition of minimum needs, so the definition of the
+unit can be referenced.
+"""
+
+# namespace prefix for field key and field name
+minimum_needs_namespace = 'minimum_needs'
+
+
+def _normalize_field_name(value):
+    """Normalizing value string used as key and field name.
+
+    :param value: String to normalize
+    :type value: str
+
+    :return: normalized string
+    :rtype: str
+    """
+    # Replace non word/letter character
+    return_value = re.sub(r'[^\w\s-]+', '', value)
+    # Replaces whitespace with underscores
+    return_value = re.sub(r'\s+', '_', return_value)
+    return return_value.lower()
+
+
+def initializes_minimum_needs_fields():
+    """Initialize minimum needs fields.
+
+    Minimum needs definitions are taken from currently used profile.
+    """
+    needs_profile = NeedsProfile()
+    needs_profile.load()
+    fields = []
+
+    for n in needs_profile.get_needs_parameters():
+        need_parameter = n
+        """:type: ResourceParameter"""
```

Comment:
I'm using pycharm typehinting. AFAIK it doesn't work with # at the moment.
Also, that's why the variable is only n, because what I used is need_parameter.
Sadly, can't typehint for variable in pycharm directly for now.